x/tokenfactory/keeper/createdenom.go

```diff
@@ -77,32 +74,43 @@ func (k Keeper) validateCreateDenom(ctx sdk.Context, creatorAddr, subdenom strin
 	return denom, nil
 }
 
-func (k Keeper) chargeFeeForDenomCreation(ctx sdk.Context, creatorAddr string) (err error) {
-	// Send creation fee to community pool
-	creationFee := k.GetParams(ctx).DenomCreationFee
-	accAddr, err := sdk.AccAddressFromBech32(creatorAddr)
-	if err != nil {
-		return errors.Wrapf(types.ErrUnableToCharge, "wrong creator address: %v", err)
-	}
-
+func (k Keeper) chargeForCreateDenom(ctx sdk.Context, creatorAddr string) (err error) {
 	params := k.GetParams(ctx)
 
-	if len(creationFee) > 0 {
+	// ORIGINAL: if DenomCreationFee is non-zero, transfer the tokens from the creator
+	// account to community pool
+	// MODIFIED: if DenomCreationFee is non-zero, transfer the tokens from the creator
+	// account to feeCollectorAddr
+	if len(params.DenomCreationFee) != 0 {
+		accAddr, err := sdk.AccAddressFromBech32(creatorAddr)
+		if err != nil {
+			return err
+		}
+		// Instead of funding community pool we send funds to fee collector addr
+		// if err := k.communityPoolKeeper.FundCommunityPool(ctx, params.DenomCreationFee, accAddr); err != nil {
+		//	return err
+		//}
+
 		feeCollectorAddr, err := sdk.AccAddressFromBech32(params.FeeCollectorAddress)
```

Comment:
since we [suggest](https://github.com/neutron-org/neutron/blob/c235d069182b1f78c29160ccf14e09f73b254fc0/x/tokenfactory/types/params.go#L33) no `params.FeeCollectorAddress` (i.e. `params.FeeCollectorAddress = ""`), this block should only be executed in case `params.FeeCollectorAddress` is not empty as well as `params.DenomCreationFee`:

```go
	if len(params.DenomCreationFee) != 0 && params.FeeCollectorAddress != "" {
```

I suppose it's reasonable to have a check in the `Params.Validate` for it, i.e. assure these params are either both defined or both undefined